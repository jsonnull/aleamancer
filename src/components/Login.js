// @flow
import React from 'react'
import styled from 'styled-components'
import Logo from './Logo'
import { fonts } from 'styles/common'

type Props = {
  email: string,
  password: string,
  onEmailChange: Function,
  onPasswordChange: Function,
  onLogin: Function
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`

const Login = styled.div`
  width: 300px;
  align-self: center;
  margin: auto;
  background: ${props => props.theme.background};
  border-radius: 5px;
  padding: 2rem;
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.3);
`

const Heading = styled.h1`
  font-family: ${fonts.heading};
  text-align: center;
  margin: 0;
  margin-bottom: 1rem;
  line-height: 3rem;
`

// FIXME: Extend input element
const Input = styled.input`
  border-radius: 0;
  background: none;
  border: none;
  padding: 0;
  border-bottom: 1px solid ${props => props.theme.borderColor};
  width: 100%;
  margin-bottom: 1rem;
`

// FIXME: Extend button element
const Button = styled.button`
  display: block;
  position: relative;
  margin: 0 auto 0;
`

export default (props: Props) => {
  return (
    <Container>
      <Login>
        <Heading>
          <Logo height="25px" />
        </Heading>
        <form onSubmit={props.onLogin}>
          <Input
            name="username"
            type="text"
            placeholder="email"
            value={props.email}
            onChange={props.onEmailChange}
          />
          <Input
            name="password"
            type="password"
            placeholder="password"
            value={props.password}
            onChange={props.onPasswordChange}
          />
          <Button onClick={props.onLogin}>Login</Button>
        </form>
      </Login>
    </Container>
  )
}
