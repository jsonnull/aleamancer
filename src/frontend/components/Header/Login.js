// @flow
import React from 'react'
import { Link } from 'react-router-dom'
import Button from './Button'

const Login = () => (
  <Link to="/login" style={{ textDecoration: 'none' }}>
    <Button>Login</Button>
  </Link>
)

export default Login
