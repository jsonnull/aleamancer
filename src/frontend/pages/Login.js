// @flow
import React from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'
import { compose, withHandlers, withState } from 'recompose'
import { performUserLogin } from 'frontend/actions'
import { fonts } from 'frontend/styles/common'
import Logo from 'frontend/components/Logo'

type Props = {
  email: string,
  password: string,
  onEmailChange: Function,
  onPasswordChange: Function,
  onLogin: Function
}

const Dialog = styled.div`
  width: 300px;
  align-self: center;
  margin: auto;
  background: ${props => props.theme.background};
  border-radius: 5px;
  padding: 2rem;
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.3);
`

const Heading = styled.h1`
  font-family: ${fonts.heading};
  text-align: center;
  margin: 0;
  margin-bottom: 1rem;
  line-height: 3rem;
`

// FIXME: Extend input element
const Input = styled.input`
  border-radius: 0;
  background: none;
  border: none;
  padding: 0;
  border-bottom: 1px solid ${props => props.theme.borderColor};
  width: 100%;
  margin-bottom: 1rem;
`

// FIXME: Extend button element
const Button = styled.button`
  display: block;
  position: relative;
  margin: 0 auto 0;
`

export const Login = (props: Props) => {
  return (
    <Dialog>
      <Heading>
        <Logo height="25px" />
      </Heading>
      <form onSubmit={props.onLogin}>
        <Input
          name="username"
          type="text"
          placeholder="email"
          value={props.email}
          onChange={props.onEmailChange}
        />
        <Input
          name="password"
          type="password"
          placeholder="password"
          value={props.password}
          onChange={props.onPasswordChange}
        />
        <Button onClick={props.onLogin}>Login</Button>
      </form>
    </Dialog>
  )
}

const enhance = compose(
  connect(null, (dispatch: Function) => ({
    login: (email: string, password: string) =>
      dispatch(performUserLogin(email, password))
  })),
  withState('email', 'setEmail', ''),
  withState('password', 'setPassword', ''),
  withHandlers({
    onEmailChange: ({ setEmail }) => (e: Object) => setEmail(e.target.value),
    onPasswordChange: ({ setPassword }) => (e: Object) =>
      setPassword(e.target.value),
    onLogin: ({ login, email, password, setPassword }) => (e: Object) => {
      e.preventDefault()
      e.stopPropagation()

      login(email, password)
      setPassword('')
    }
  })
)

export default enhance(Login)
