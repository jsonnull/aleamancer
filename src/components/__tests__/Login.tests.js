// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import Login from '../Login'

describe('Login component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Login
          email="test@example.com"
          password="test"
          onEmailChange={() => {}}
          onPasswordChange={() => {}}
          onLogin={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
