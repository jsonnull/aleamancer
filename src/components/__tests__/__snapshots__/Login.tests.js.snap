// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<div
  className="sc-bwzfXH fYVOJF"
>
  <div
    className="sc-htpNat iwoucR"
  >
    <h1
      className="sc-bxivhb kBnUkJ"
    >
      <div
        className="sc-bdVaJa cNYobl"
        dangerouslySetInnerHTML={
          Object {
            "__html": Object {},
          }
        }
        height="25px"
      />
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="sc-ifAKCX frYxG"
        name="username"
        onChange={[Function]}
        placeholder="email"
        type="text"
        value="test@example.com"
      />
      <input
        className="sc-ifAKCX frYxG"
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value="test"
      />
      <button
        className="sc-EHOje hZyWZt"
        onClick={[Function]}
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
